tfjs-backend-webgpu/src/kernels/Cumprod.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util, Cumprod, CumprodAttrs, CumprodInputs, KernelConfig, KernelFunc, TensorInfo} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {CumOpType, CumProgram} from '../cum_webgpu';
+
+import {identity} from './Identity';
+import {transpose} from './Transpose';
+
+export function cumprod(
```

Comment:
`cumprod` and `cumsum` has many duplicated code. Is it possible to call a common function like `Cum_impl` to reduce the duplication?

Reply:
Done